packages/strapi-plugin-i18n/config/functions/migrations/field.js

```diff
@@ -0,0 +1,184 @@
+'use strict';
+
+const { difference, pick, orderBy, prop, intersection } = require('lodash/fp');
+const { getService } = require('../../../utils');
+
+const BATCH_SIZE = 1000;
+
+const shouldBeProcesseed = processedLocaleCodes => entry => {
+  return (
+    entry.localizations.length > 1 &&
+    intersection(entry.localizations.map(prop('locale')), processedLocaleCodes).length === 0
+  );
+};
+
+const getUpdates = ({ entriesToProcess, formatUpdate, locale, attributesToMigrate }) => {
+  return entriesToProcess.reduce((updates, entry) => {
```

Comment:
we can optimize perf with a for loop & pushing into an array to avoid the heavy concats